visidata/save.py

```diff
@@ -87,7 +87,11 @@ def getDefaultSaveName(sheet):
     if isinstance(src, Path):
         if sheet.options.is_set('save_filetype', sheet):
             # if save_filetype is over-ridden from default, use it as the extension
+            if src.given == '-':
+                return f'stdin.{sheet.options.save_filetype}'
```

Comment:
I think we should use the extension for stdin always.  The `with_suffix` will replace the extension deliberately if the savetype is set deliberately, but the point of the branch is to use the original extension otherwise, so you can easily save to the file you loaded from.